internal/gribi/gribi.go

```diff
@@ -63,6 +63,25 @@ func (c *Client) Fluent(t testing.TB) *fluent.GRIBIClient {
 	return c.fluentC
 }
 
+// OptionalParam is used to pass additional parameters to gribi APIs
+type OptionalParam interface {
+	// IsTestOpt marks the TestOpt as implementing this interface.
+	IsOptionalParam()
+}
+
+// BKNHG is used to pass the backup nexthop group for adding next hop group
+type BKNHG struct {
```

Comment:
Maybe we don't need an interface, just make this an NHGOptions struct, rename id to BackupNHG, and only apply this if id > 0?

Reply:
The reason I am using an interface is that we may need to extend current function parameters, add other functions in gribi file in the future and I want to avoid changing calls on the caller's side. I can still make the change if you prefer struct.

Reply:
This should not require the caller to change anything when you add new fields:

```
client := &gribi.Client{...}
client.AddNHG(t, 42, ..., &gribi.NHGOptions{BackupNHG: 10})
```

Besides, gribi is an internal package, so the usage is limited to this repo which we have full control over. Someone who introduces a breaking change could see this in pull request checks and will be able to fix it.